store/sqlstore/plugin_store.go

```diff
@@ -100,15 +110,18 @@ func (ps SqlPluginStore) CompareAndSet(kv *model.PluginKeyValue, oldValue []byte
 			}
 		}
 	} else {
+		currentTime := model.GetMillis()
+
 		// Update if oldValue is not nil
 		updateResult, err := ps.GetMaster().Exec(
-			`UPDATE PluginKeyValueStore SET PValue = :New, ExpireAt = :ExpireAt WHERE PluginId = :PluginId AND PKey = :Key AND PValue = :Old`,
+			`UPDATE PluginKeyValueStore SET PValue = :New, ExpireAt = :ExpireAt WHERE PluginId = :PluginId AND PKey = :Key AND PValue = :Old AND (ExpireAt = 0 OR ExpireAt > :CurrentTime)`,
```

Comment:
This will leave the expired keys in the store though, right?

Reply:
That's correct -- we do have a daily background job that prunes expired keys, but from a semantic perspective, the expired key should be treated as if it never existed in the first place, failing the `KVCompareAndSet` when `oldValue != nil`.